frontend/src/metabase/public/components/widgets/EmbedModal.tsx

```diff
@@ -0,0 +1,77 @@
+import { titleize } from "inflection";
+import { useState } from "react";
+import { t } from "ttag";
+import { useSelector } from "metabase/lib/redux";
+import { getApplicationName } from "metabase/selectors/whitelabel";
+import { Icon } from "metabase/core/components/Icon";
+import Modal from "metabase/components/Modal";
+import type { WindowModalProps } from "metabase/components/Modal/WindowModal";
+import { Box, Center } from "metabase/ui";
+import { EmbedTitleLabel } from "./EmbedModal.styled";
+
+type EmbedModalStep = "application" | null;
+
+const EmbedTitle = ({
+  type,
+  onClick = undefined,
+}: {
+  type: string;
+  onClick?: () => void;
+}) => (
+  <a className="flex align-center" onClick={onClick}>
+    <EmbedTitleLabel>{t`Sharing`}</EmbedTitleLabel>
+    {type && <Icon name="chevronright" className="mx1 text-medium" />}
+    {type}
+  </a>
+);
+
+export const EmbedModal = ({
+  children,
+  isOpen,
+  onClose,
+  ...modalProps
+}: {
+  isOpen?: boolean;
+  onClose: () => void;
+  children: ({
+    embedType,
+    setEmbedType,
+  }: {
+    embedType: EmbedModalStep;
+    setEmbedType: (type: EmbedModalStep) => void;
+  }) => JSX.Element;
+} & WindowModalProps) => {
+  const [embedType, setEmbedType] = useState<EmbedModalStep>(null);
+  const applicationName = useSelector(getApplicationName);
```

Comment:
@WiNloSt let me know if this is how it should look 😄

Reply:
This is correct!

Reply:
![](https://i.shipit.today/bobbyawesome)